stackrox/stackrox
# ROX-21196: Add new roxctl flag for namespace

## Description
Add a `--namespace` flag to `roxctl deployment check` and pass the value to the the `DeployYAMLDetectionRequest`. Handling the new value in that request on the Sensor side is **not** implemented yet

## Checklist

If any of these don't apply, please comment below.

## Testing Performed

No automatic tests added, e2e-tests once sensor side is ready too would be reasonable.
Manually tested the warnings

```
jschnath-mac:yaml jschnath$ ../../../../bin/darwin_amd64/roxctl -e 127.0.0.1:8000 deployment check --file nginx.yaml --namespace stackrox
WARN:	Namespace has been set, but cluster remains unset. Setting the namespace only makes sense when also setting the cluster with '--cluster [CLUSTER]'.

```
```
jschnath-mac:yaml jschnath$ ../../../../bin/darwin_amd64/roxctl -e 127.0.0.1:8000 deployment check --file nginx.yaml --cluster 123
WARN:	No custom namespace provided. Any deployments without an explicit namespace will be defaulted to "default". Overwrite the default with --namespace/-n [NAMESPACE]
```
